src/Adapter/Customer/Group/QueryHandler/GetCustomerGroupForEditingHandler.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Adapter\Customer\Group\QueryHandler;
+
+use Group as CustomerGroup;
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Adapter\Customer\Group\Repository\GroupRepository;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Exception\GroupNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Query\GetCustomerGroupForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\QueryHandler\GetCustomerGroupForEditingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\QueryResult\EditableCustomerGroup;
+
+class GetCustomerGroupForEditingHandler implements GetCustomerGroupForEditingHandlerInterface
+{
+    /**
+     * @var GroupRepository
+     */
+    private $customerGroupRepository;
+
+    public function __construct(GroupRepository $customerGroupRepository)
+    {
+        $this->customerGroupRepository = $customerGroupRepository;
+    }
+
+    /**
+     * @param GetCustomerGroupForEditing $query
+     *
+     * @throws GroupNotFoundException
+     *
+     * @return EditableCustomerGroup
+     */
+    public function handle(GetCustomerGroupForEditing $query): EditableCustomerGroup
+    {
+        $customerGroupId = $query->getCustomerGroupId();
+        $this->customerGroupRepository->assertGroupExists($customerGroupId);
+
+        $customerGroup = new CustomerGroup($customerGroupId->getValue());
```

Comment:
get it from repository instead of manually loading here

Reply:
```suggestion
        $customerGroup = $this->customerGroupRepository->get($customerGroupId)
```
@jolelievre im just unsure about shop.
There is ps_group_shop which only contains association, so shouldn't we care if there is no association in certain shop?
like should we instead do `$repo->get($groupId, $shopId)  and there is no group for that shop we should throw ShopAssociationException, no?

Reply:
oh its actually the same concept as here https://github.com/PrestaShop/PrestaShop/pull/31502#issuecomment-1456277301, so no need shopId, nevermind.